### PR TITLE
Deploy script for the protocol and plugin upgrades

### PR DESCRIPTION
**Pull request by Michael-A-Heuer**
## Description

Note, that @mathewmeconry tested the previous deploy scripts with tenderly and that tests are worked on by @Rekard0 in [Task OS-441](https://aragonassociation.atlassian.net/browse/OS-441).

Task: [OS-416](https://aragonassociation.atlassian.net/browse/OS-416)

## Type of change

- deployment preparation

[OS-416]: https://aragonassociation.atlassian.net/browse/OS-416?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/contracts/deploy/update/to_v1.3.0/20_TokenVoting_Plugin.ts [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      deployer.address,
+      await tokenVotingRepo.MAINTAINER_PERMISSION_ID(),
+      '0x00'
+    )
+  ) {
+    console.log(`Deployer has permission to install new TokenVoting version`);
+    const tx = await tokenVotingRepo
+      .connect(deployer)
+      .createVersion(
+        1,
```
Comment (brickpop) [inline packages/contracts/deploy/update/to_v1.3.0/20_TokenVoting_Plugin.ts:52]: Very minor, but I would create a constant like `TARGET_RELEASE` at the top, factoring out all occurrences of this

File: packages/contracts/deploy/update/to_v1.3.0/30_AddresslistVoting_Plugin.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      '0x00'
+    )
+  ) {
+    console.log(
+      `Deployer has permission to install new AddresslistVoting version`
+    );
+    const tx = await addresslistVotingRepo
+      .connect(deployer)
+      .createVersion(
+        1,
```
Comment (brickpop) [inline packages/contracts/deploy/update/to_v1.3.0/30_AddresslistVoting_Plugin.ts:54]: Same as before

File: packages/contracts/deploy/update/to_v1.3.0/20_TokenVoting_Plugin.ts (hunk 1 above)
Comment (Michael-A-Heuer) [reply packages/contracts/deploy/update/to_v1.3.0/20_TokenVoting_Plugin.ts]: Added in bd20b893734d9afa715bf68128e4514ffeed67ea

File: packages/contracts/deploy/update/to_v1.3.0/30_AddresslistVoting_Plugin.ts (hunk 2 above)
Comment (Michael-A-Heuer) [reply packages/contracts/deploy/update/to_v1.3.0/30_AddresslistVoting_Plugin.ts]: Added in bd20b893734d9afa715bf68128e4514ffeed67ea

**Review by brickpop — DISMISSED** [review]

**Review by Rekard0 — COMMENTED** [review]
could u add a tag to all files, expressing the update
something like `update_to_1_3_0` or what ever tag u prefer

File: packages/contracts/deploy/update/to_v1.3.0/01_DAOFactory.ts [hunk 3]
```diff
@@ -23,12 +24,13 @@ const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
...
 
   const deployResult = await deploy('DAOFactory', {
+    contract: daoFactoryArtifact,
     from: deployer.address,
     args: [daoRegistryAddress, pluginSetupProcessorAddress],
     log: true,
   });
 
-  const daoInterface = DAO__factory.createInterface();
+  const daoInterface = new ethers.utils.Interface(daoFactoryArtifact.abi);
```
[the remark this answers is not in the document]
Comment (Michael-A-Heuer) [reply packages/contracts/deploy/update/to_v1.3.0/01_DAOFactory.ts]: Fixed in 021c1de3dd83acb4bafa01b8ff1aba4aace85235

**Review by brickpop — APPROVED** [review]

**Review by Rekard0 — APPROVED** [review]
LGTM

**Pull request merged**